demos/Shrine/Shrine/ShrineHeaderContentView.swift

```diff
@@ -85,118 +92,56 @@ class ShrineHeaderContentView: UIView, UIScrollViewDelegate {
-    imageView2.autoresizingMask = .flexibleHeight
-    (secondPage as AnyObject).addSubview(imageView2)
-    let url2 = URL(string: ShrineData.baseURL + "backpack.png")
-    remoteImageService.fetchImageAndThumbnail(from: url2) { (image:UIImage?,
-      thumbnailImage:UIImage?) -> Void in
-      DispatchQueue.main.async(execute: {
-        self.imageView2.image = image;
-        self.imageView2.setNeedsDisplay()
-      })
-    }
+  }
 
-    label2.font = fontAbril
-    label2.textColor = textColor
-    label2.lineBreakMode = .byWordWrapping
-    label2.numberOfLines = 2
-    label2.attributedText = attributedString("Best gift for \nthe traveler",
-      lineHeightMultiple: 0.8)
-    (secondPage as AnyObject).addSubview(label2)
-
-    labelDesc2.lineBreakMode = .byWordWrapping
-    labelDesc2.numberOfLines = 2
-    labelDesc2.font = fontHelvetica
-    labelDesc2.textColor = descColor
-    labelDesc2.attributedText = attributedString(descString, lineHeightMultiple: 1.2)
-    (secondPage as AnyObject).addSubview(labelDesc2)
-
-    cyanBox2.backgroundColor = cyanBoxColor
-    (secondPage as AnyObject).addSubview(cyanBox2)
-
-    let inkOverlay2 = ShrineInkOverlay(frame: (secondPage as AnyObject).bounds)
-    inkOverlay2.autoresizingMask = [.flexibleWidth, .flexibleHeight]
-    (secondPage as AnyObject).addSubview(inkOverlay2)
-
-    imageView3.contentMode = UIViewContentMode.scaleAspectFill
-    imageView3.autoresizingMask = .flexibleHeight
-    (thirdPage as AnyObject).addSubview(imageView3)
-    let url3 = URL(string: ShrineData.baseURL + "heels.png")
-    remoteImageService.fetchImageAndThumbnail(from: url3) { (image:UIImage?,
-      thumbnailImage:UIImage?) -> Void in
-      DispatchQueue.main.async(execute: {
-        self.imageView3.image = image;
-        self.imageView3.setNeedsDisplay()
-      })
-    }
+  func addPageContent() {
+    let firstPage = pages[0]
+    let secondPage = pages[1]
+    let thirdPage = pages[2]
 
-    label3.font = fontAbril
-    label3.textColor = textColor
-    label3.lineBreakMode = .byWordWrapping
-    label3.numberOfLines = 2
-    label3.attributedText = attributedString("Better \nwearing heels", lineHeightMultiple: 0.8)
-    (thirdPage as AnyObject).addSubview(label3)
-
-    labelDesc3.lineBreakMode = .byWordWrapping
-    labelDesc3.numberOfLines = 2
-    labelDesc3.font = fontHelvetica
-    labelDesc3.textColor = descColor
-    labelDesc3.attributedText = attributedString(descString, lineHeightMultiple: 1.2)
-    (thirdPage as AnyObject).addSubview(labelDesc3)
-
-    cyanBox3.backgroundColor = cyanBoxColor
-    (thirdPage as AnyObject).addSubview(cyanBox3)
-
-    let inkOverlay3 = ShrineInkOverlay(frame: (thirdPage as AnyObject).bounds)
-    inkOverlay3.autoresizingMask = [.flexibleWidth, .flexibleHeight]
-    (thirdPage as AnyObject).addSubview(inkOverlay3)
+    addPage(page: firstPage as! UIView, imageView: imageView, label: label, labelDesc: labelDesc,
```

Comment:
Great improvement! I guess the next step would be to pull this information into a model class.